pkg/operator/discover/discover.go

```diff
@@ -130,6 +137,7 @@ func (d *Discover) createDiscoverDaemonSet(ctx context.Context, namespace, disco
 									ReadOnly:  true,
 								},
 							},
+							Resources: discoverDaemonResources[0].Resource,
```

Comment:
Before accessing the index check, the length is not 0.

Reply:
THanks for this suggestion, updated